test-suite/src/test/scala/org/scalajs/testsuite/niocharset/UTF8Test.scala

```diff
@@ -87,6 +82,10 @@ object UTF8Test extends BaseCharsetTest(StandardCharsets.UTF_8) {
       testDecode(bb"f0")(Malformed(1))
       testDecode(bb"f0 90")(Malformed(2))
       testDecode(bb"f0 90 80")(Malformed(3))
+      // at the end of the buffer - #1537
+      testDecode(bb"c0")(Malformed(1))
+      testDecode(bb"e1 41")(Malformed(1), cb"A")
+      testDecode(bb"e1 80 42")(Malformed(2), cb"B")
```

Comment:
Before someone asks, [the bug report to OpenJDK](https://bugs.openjdk.java.net/browse/JDK-6795537) incorrectly says that the expected output for this one is `Malformed(1)`. It is in fact `Malformed(2)`, as can be seen by the following reproduction in the REPL:
```scala
scala> decoder.decode(ByteBuffer.wrap(Array(0xe1.toByte, 0x80.toByte, 0x42.toByte)), CharBuffer.allocate(10), true)
res0: java.nio.charset.CoderResult = MALFORMED[2]
```